src/dmd/expressionsem.d

```diff
@@ -11508,3 +11484,55 @@ private bool checkFunctionAttributes(Expression exp, Scope* sc, FuncDeclaration
         return error;
     }
 }
+
+/*******************************
+ * Helper function for `getRightThis()`.
+ * Gets `this` of the next outer aggregate.
+ * Params:
+ *      loc = location to use for error messages
+ *      sc = context
+ *      s = the parent symbol of the existing `this`
+ *      ad = struct or class we need the correct `this` for
+ *      e1 = existing `this`
+ *      t = type of the existing `this`
+ *      var = the specific member of ad we're accessing
+ *      flag = if true, return `null` instead of throwing an error
```

Comment:
I think `flag` could have been omitted from this PR, but I see where this is going.

Reply:
why do you thinks so? it is used to keep the current behavior untouched. in `expressionsem.d` an error message is generated but not in `typesem.d`.

Reply:
Ah, you are right.  My apologies.